tutorials/images/Images_Tutorial.ipynb

```diff
@@ -0,0 +1,444 @@
+    "        \n",
+    "    P = PrimitiveObject()\n",
+    "    P.save_primitive_matrix(primitive_mtx)\n",
+    "    return P"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Assign and Name Primitives**\n",
+    "We assign the primitives and name them according to the variables we will use to refer to them in the heuristic functions we develop next. \n",
+    "\n",
+    "For example, `primitive_mtx[i,0]` is referred to as `has_human`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "P = create_primitives(loader)\n",
+    "primitive_names = ['has_human', 'has_road', 'has_cars', 'has_bike', \n",
+    "                   'bike_human_distance', 'bike_human_size', 'bike_human_num']"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Heuristic Functions\n",
+    "We now develop heuristic functions that take different primitives in as inputs and apply a label based on the value of those primitives. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def LF_street(has_human, has_road):\n",
+    "    if has_human >= 1: \n",
+    "        if has_road >= 1:\n",
+    "            return 1\n",
+    "        else:\n",
+    "            return -1\n",
+    "    return 0\n",
+    "\n",
+    "def LF_vehicles(has_human, has_cars):\n",
+    "    if has_human >= 1: \n",
+    "        if has_cars >= 1:\n",
+    "            return 1\n",
+    "        else:\n",
+    "            return -1\n",
+    "    return 0\n",
+    "\n",
+    "def LF_distance(has_human, has_bike, bike_human_distance):\n",
+    "    if has_human >= 1:\n",
+    "        if has_bike >= 1: \n",
+    "            if bike_human_distance <= np.sqrt(8):\n",
+    "                return 1\n",
+    "            else:\n",
+    "                return 0\n",
+    "    else:\n",
+    "        return -1\n",
+    "    \n",
+    "def LF_size(has_human, has_bike, bike_human_size):\n",
+    "    if has_human >= 1:\n",
+    "        if has_bike >= 1: \n",
+    "            if bike_human_size <= 1000:\n",
+    "                return -1\n",
+    "            else:\n",
+    "                return 0\n",
+    "    else:\n",
+    "        return -1\n",
+    "    \n",
+    "    \n",
+    "def LF_number(has_human, has_bike, bike_human_num):\n",
+    "    if has_human >= 1:\n",
+    "        if has_bike >= 1: \n",
+    "            if bike_human_num >= 2:\n",
+    "                return 1\n",
+    "            if bike_human_num >= 1:\n",
+    "                return 0\n",
+    "            if bike_human_num >= 0:\n",
+    "                return 1\n",
+    "    else:\n",
+    "        return -1"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Assign Heuristic Functions**\n",
```

Comment:
Say `Labeling functions / LFs` to be consistent with earlier in the tutorial & rest of Snorkel code

Reply:
(comment applies throughout!)